Allow usage with private github instances

Trying to use this project to build a bot for our private github instance.

It looks like it is currently impossible to specify a specific github host.

I followed the setup guide and when running the app in dev mode, I get the following error (stripped some stacktrace elements):
```
Exception: java.lang.IllegalStateException: Unable to create a GitHub token for the installation 959
        at io.quarkiverse.githubapp.runtime.github.GitHubService$CreateInstallationToken.load(GitHubService.java:117)
        at io.quarkiverse.githubapp.runtime.github.GitHubService$CreateInstallationToken.load(GitHubService.java:105)
        at com.github.benmanes.caffeine.cache.LocalLoadingCache.lambda$newMappingFunction$2(LocalLoadingCache.java:141)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2413)
        at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1916)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2411)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2394)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108)
        at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:54)
        at io.quarkiverse.githubapp.runtime.github.GitHubService.createInstallationClient(GitHubService.java:93)
        ... more
Caused by: org.kohsuke.github.GHFileNotFoundException: https://api.github.com/app {"message":"Integration must generate a public key","documentation_url":"https://docs.github.com/rest"}
        at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:480)
        at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:407)
        at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:355)
        at org.kohsuke.github.Requester.fetch(Requester.java:76)
        at org.kohsuke.github.GitHub.getApp(GitHub.java:1109)
        at io.quarkiverse.githubapp.runtime.github.GitHubService$CreateInstallationToken.load(GitHubService.java:111)
        ... 61 more
Caused by: java.io.FileNotFoundException: https://api.github.com/app
        at org.kohsuke.github.extras.okhttp3.ObsoleteUrlFactory$OkHttpURLConnection.getInputStream(ObsoleteUrlFactory.java:490)
        at org.kohsuke.github.extras.okhttp3.ObsoleteUrlFactory$DelegatingHttpsURLConnection.getInputStream(ObsoleteUrlFactory.java:1204)
        at org.kohsuke.github.GitHubHttpUrlConnectionClient$HttpURLConnectionResponseInfo.bodyStream(GitHubHttpUrlConnectionClient.java:196)
        at org.kohsuke.github.GitHubResponse$ResponseInfo.getBodyAsString(GitHubResponse.java:314)
        at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:92)
        at org.kohsuke.github.Requester.lambda$fetch$1(Requester.java:76)
        at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:449)

```

My guess is that the issue lies in the `GitHubService` class, in the `createInstallationClient` method where we could inject an endpoint url and call the `GithubBuilder.withEndpoint()` method.

I will try to make a PR for this.

--- a/docs/modules/ROOT/pages/includes/quarkus-github-app.adoc
+++ b/docs/modules/ROOT/pages/includes/quarkus-github-app.adoc
@@ -55,6 +55,16 @@ The Smee.io proxy URL used when testing locally.
 |
 
 
+a| [[quarkus-github-app_quarkus.github-app.instance-endpoint]]`link:#quarkus-github-app_quarkus.github-app.instance-endpoint[quarkus.github-app.instance-endpoint]`
+
+[.description]
+--
+The GitHub instance endpoint. 
+ If not specified, defaults to the public github.com instance.
+--|string 
+|
+
+
 a| [[quarkus-github-app_quarkus.github-app.debug.payload-directory]]`link:#quarkus-github-app_quarkus.github-app.debug.payload-directory[quarkus.github-app.debug.payload-directory]`
 
 [.description]

--- a/runtime/src/main/java/io/quarkiverse/githubapp/runtime/config/GitHubAppRuntimeConfig.java
+++ b/runtime/src/main/java/io/quarkiverse/githubapp/runtime/config/GitHubAppRuntimeConfig.java
@@ -47,7 +47,9 @@ public class GitHubAppRuntimeConfig {
     public Optional<String> webhookProxyUrl;
 
     /**
-     * The github instance endpoint.
+     * The GitHub instance endpoint.
+     * <p>
+     * If not specified, defaults to the public github.com instance.
      */
     @ConfigItem
     public Optional<String> instanceEndpoint;

--- a/runtime/src/main/java/io/quarkiverse/githubapp/runtime/github/GitHubService.java
+++ b/runtime/src/main/java/io/quarkiverse/githubapp/runtime/github/GitHubService.java
@@ -92,7 +92,7 @@ public class GitHubService {
     private GitHub createInstallationClient(Long installationId) throws IOException {
         CachedInstallationToken installationToken = installationTokenCache.get(installationId);
 
-        final var gitHubBuilder = new GitHubBuilder().withConnector(okhttpConnector)
+        final GitHubBuilder gitHubBuilder = new GitHubBuilder().withConnector(okhttpConnector)
                 .withAppInstallationToken(installationToken.getToken());
 
         gitHubAppRuntimeConfig.instanceEndpoint.ifPresent(gitHubBuilder::withEndpoint);

